Create "Networks" page in sc basics (#380)
polkadot-developers/polkadot-docs: polkadot-protocol/smart-contract-basics/networks.md

---
title: Networks for Asset Hub Smart Contracts
description: Explore the available networks for smart contract development on Asset Hub, including Westend Asset Hub, Kusama Asset Hub, and Polkadot Asset Hub.
---

# Networks

## Introduction

> Smart contract development in the Polkadot ecosystem is primarily facilitated through Asset Hub, which provides robust smart contract functionality across multiple networks. Whether you're testing new contracts or deploying to production, Asset Hub offers several network environments tailored for each stage of development. Developers can thoroughly test, iterate, and validate their smart contracts from local testing environments to production networks like Polkadot Asset Hub.

Reviewer: ```suggestion
Smart contract development in the Polkadot ecosystem is primarily facilitated through Asset Hub, which provides smart contract functionality across multiple networks. Whether you're testing new contracts or deploying to production, Asset Hub offers several network environments tailored for each stage of development. Developers can thoroughly test, iterate, and validate their smart contracts from local testing environments to production networks like Polkadot Asset Hub.
```

This guide will introduce you to the various networks available (and incoming) for smart contract development and explain how they fit into the development workflow.

## Network Overview

Smart contract development on Asset Hub follows a structured process to ensure new contracts and upgrades are rigorously tested before deployment on production networks. The progression follows a well-defined path, starting from local environments and advancing through TestNets, ultimately reaching the Polkadot Asset Hub. The diagram below outlines the typical progression of the smart contract development cycle:

``` mermaid
flowchart LR
    id1[Local] --> id2[Westend Asset Hub] --> id4[Kusama Asset Hub] --> id5[Polkadot Asset Hub]
    id1[Local] --> id3[Paseo Asset Hub] --> id5[Polkadot Asset Hub]
```

This flow ensures developers can thoroughly test and iterate their smart contracts without risking real tokens or affecting production networks. Development tools and various TestNets make it easier to experiment safely before releasing it to production.

> A typical journey through the smart contract development process might look like this:

Reviewer: I don't think this path is ok. Once the smart contracts functionality is stable, Paseo will probably be the best place to test smart contracts for developers as it will mimic Polkadot AssetHub. Westend will be used by core developers to test and improve the runtime and it may not be as stable as needed by smart contract developers

Author: Working on it!

Author: fixed in [798da78](https://github.com/polkadot-developers/polkadot-docs/pull/380/commits/798da78643047c659986e47abff7a666ba30d779)

- [**Ethereum RPC proxy**](https://paritytech.github.io/polkadot-sdk/master/pallet_revive_eth_rpc/index.html){target=\_blank} - bridges Ethereum-compatible tools with the Polkadot SDK-based network. It enables seamless integration with popular development tools like MetaMask and Remix IDE. The purpose of this component is to translate Ethereum RPC calls into Substrate format

## Test Networks

The following test networks provide controlled environments for testing smart contracts. TestNet tokens are available from the [Polkadot faucet](https://faucet.polkadot.io/){target=\_blank}.

### Westend Asset Hub

Westend Asset Hub is the primary permanent TestNet for smart contract development. It provides a stable environment for testing contracts without using real tokens. The network maintains the same features and capabilities as the production Asset Hub, making it an ideal environment for thorough testing before moving to production networks.

### Paseo Asset Hub

> The upcoming Paseo Asset Hub will serve as a community-managed TestNet specifically designed for smart contract development. It will mirror Asset Hub's runtime and provide developers with an additional environment for testing their contracts before deployment to production networks.

Reviewer 2: ```suggestion
The upcoming Paseo Asset Hub will be a community-managed TestNet designed specifically for smart contract development. It will mirror Asset Hub's runtime and provide developers with an additional environment for testing their contracts before deployment to production networks.
```
